darts/dataprocessing/transformers/midas.py

```diff
@@ -0,0 +1,328 @@
+        low_freq_series_df.index = low_index_datetime.to_period()
+        high_freq_series_df = low_freq_series_df.resample(high_freq_period).last()
+
+        # make sure the extension of the index matches the original index
+        if "End" in str(series.freq):
+            args_to_timestamp = {"freq": high_freq_period}
+        else:
+            args_to_timestamp = {"how": "start"}
+        high_index_datetime = high_freq_series_df.index.to_timestamp(
+            **args_to_timestamp
+        )
+
+        raise_if_not(
+            low_freq_series_df.shape[0] < high_freq_series_df.shape[0],
+            f"The target conversion should go from a high to a "
+            f"low frequency, instead the targeted frequency is "
+            f"{rule}, while the original frequency is {high_freq_datetime}.",
+            logger,
+        )
+
+        # if necessary, expand the original series
+        if len(high_index_datetime) > series_df.shape[0]:
+            series_df = pd.DataFrame(
+                np.nan, index=high_index_datetime, columns=series_copy_df.columns
+            )
+            series_df.loc[series_copy_df.index, :] = series_copy_df.values
+
+        # make multiple low frequency columns out of the high frequency column(s)
+        midas_df = _create_midas_df(
+            series_df=series_df,
+            low_index_datetime=low_index_datetime,
+        )
+
+        # back to TimeSeries
+        midas_ts = TimeSeries.from_dataframe(
+            midas_df,
+            static_covariates=series.static_covariates,
+        )
+        if strip:
+            midas_ts = midas_ts.strip()
+
+        # components: comp0_0, comp1_0, comp0_1, comp1_1, ...
+        return midas_ts
+
+    @staticmethod
+    def ts_inverse_transform(
+        series: TimeSeries,
+        params: Mapping[str, Any],
+    ) -> TimeSeries:
+        """
+        Transforms series back to high frequency by retrieving the original high frequency and reshaping the values.
+        When converting to/from anchorable offset [1]_, the original time index is not garanteed to be restored.
+
+        Steps:
+            (1) Reshape the values to flatten the components introduced by the transform
+            (2) Eliminate the rows filled with NaNs, to facilitate time index adjustment
+            (3) Retrieve the original components name
+            (4) When applicable, shift the time index start back in time
+            (5) Generate a new time index with the high frequency
+
+        References
+        ----------
+        .. [1] https://pandas.pydata.org/pandas-docs/stable/user_guide/timeseries.html#anchored-offsets
+        """
+        MIDAS._verify_series(series)
+        high_freq_name = params["fitted"]["high_freq_name"]
+        orig_ts_start_time = params["fitted"]["start"]
+        orig_ts_end_time = params["fitted"]["end"]
+
+        # retrieve the number of component introduced by midas
+        n_midas_components = int(series.components[-1].split("_")[-1]) + 1
```

Comment:
We are relying on the time indexes (low and high frequencies) generated by pandas.resample to find the multiple, I will try to think of a more efficient way but it will probably not be straightforward.